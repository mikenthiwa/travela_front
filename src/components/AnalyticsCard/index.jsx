import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

const AnalyticsCard = ({ title, children }) => (
  <div className="analytics-card">
    <h3 className="analytics-card__title">{title}</h3>
    {children}
  </div>
);

AnalyticsCard.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  title: PropTypes.string.isRequired
};

export default AnalyticsCard;
