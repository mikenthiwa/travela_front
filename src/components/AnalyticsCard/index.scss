.analytics-card {
  height: 225px;
  padding: 0 30px 30px 30px;
  font-size: 15px;
  color: #4F4F4F;
  line-height: 20px;
  box-sizing: border-box;
  font-family: "DIN Pro";
  background-color: #FFFFFF;
  width: 100% !important;
  border-radius: 2px;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);

  &:not(:last-child) {
    margin-right: 10px;
  }

  .recharts-surface {
    z-index: 1;
  }

  &__title {
    font-size: 15px;
    line-height: 20px;
    margin: 0;
    padding: 20px 0 10px;
    font-family: "DIN Pro";
    height: 40px;
  }

  &__stats {
    height: 136px;
    padding: 30px;
    display: flex;
    box-sizing: border-box;
    background-color: #F8F8F8;
    justify-content: space-between;
    width: 100%;
    border-radius: 2px;
    &.green {
      background-color: #23B57B;
      & h4 {
        color: #fff;
      }
    }
    &.brown-orange {
      background-color: #FFAF30;
      & h4 {
        color: #fff;
      }
    }
  }

  &__stat {
    box-sizing: border-box;
    display: flex;
    & h4 {
      line-height: 0;
      align-self: flex-end;
      color: #3359DB;
      font-size: 54px;
      font-family: 'DIN Pro';

    }
  }
}

.label {
  font-size: 12px;
  font-family: DIN Pro;
  @media (max-width: 479px) {
    display: block !important;
  }
}

.chart-data {
  display: flex;
  justify-content: center;
  align-content: center;
}

.recharts-wrapper{
  display: flex;
  justify-content: center;
  width: auto !important;
}

@import 'responsiveness';
