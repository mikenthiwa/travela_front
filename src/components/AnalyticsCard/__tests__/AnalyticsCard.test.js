import React from 'react';
import { shallow } from 'enzyme';
import AnalyticsCard from '..';

const props = {
  stats: 30,
  color: 'green',
  title: 'Test title',
  children: []
};

describe('<Dashboard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AnalyticsCard {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
