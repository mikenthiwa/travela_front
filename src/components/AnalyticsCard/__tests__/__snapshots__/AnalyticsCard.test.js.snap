// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders correctly 1`] = `
<div
  className="analytics-card"
>
  <h3
    className="analytics-card__title"
  >
    Test title
  </h3>
</div>
`;
